ledger-core/virtual/integration/constructor_test.go

```diff
@@ -548,3 +553,53 @@ func TestVirtual_CallConstructorFromConstructor(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestVirtual_Constructor_WrongConstructorName(t *testing.T) {
+	t.Log("C4977")
+
+	var (
+		mc = minimock.NewController(t)
+	)
+
+	server, ctx := utils.NewServer(nil, t)
+	defer server.Stop()
+
+	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+	var (
+		isolation = contract.ConstructorIsolation()
+		outgoing  = server.RandomLocalWithPulse()
+		objectRef = reference.NewSelf(outgoing)
+	)
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+		Callee:         testwallet.GetClass(),
+		CallSiteMethod: "NotExistingConstructorName",
+		CallOutgoing:   outgoing,
+		Arguments:      insolar.MustSerialize([]interface{}{}),
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+		require.Equal(t, res.Callee, objectRef)
```

Comment:
please, fix require syntax:
`func Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{}) {`
move expected and actual values

Reply:
Good point, done.